pkg/controller/master/nodedrain/nodedrain_controller.go

```diff
@@ -77,121 +77,152 @@ func (ndc *ControllerHandler) OnNodeChange(_ string, node *corev1.Node) (*corev1
 	if node == nil || node.DeletionTimestamp != nil {
 		return node, nil
 	}
-	_, ok := node.Annotations[drainhelper.DrainAnnotation]
 
+	_, ok := node.Annotations[drainhelper.DrainAnnotation]
 	if !ok {
 		return node, nil
 	}
 
 	_, forced := node.Annotations[drainhelper.ForcedDrain]
 
-	if ok {
-		// still running a check in the background to avoid maintenance issues when using object annotations
-		// directly
-		err := drainhelper.DrainPossible(ndc.nodeCache, node)
-		if err != nil {
-			return node, err
+	logrus.WithFields(logrus.Fields{
+		"node_name": node.Name,
+		"forced":    forced,
+	}).Info("Attempting to place node in maintenance mode")
+
+	// still running a check in the background to avoid maintenance issues when using object annotations
+	// directly
+	err := drainhelper.DrainPossible(ndc.nodeCache, node)
+	if err != nil {
+		if errors.Is(err, drainhelper.ErrNodeDrainNotPossible) {
+			logrus.WithFields(logrus.Fields{
+				"node_name": node.Name,
+			}).Warnf("Draining is not possible: %v", err)
+
+			nodeCopy := node.DeepCopy()
+
+			// Cleanup annotations to avoid recurring approaches to drain the node.
+			delete(nodeCopy.Annotations, ctlnode.MaintainStatusAnnotationKey)
+			delete(nodeCopy.Annotations, drainhelper.DrainAnnotation)
+			delete(nodeCopy.Annotations, drainhelper.ForcedDrain)
+
+			_, errUpdate := ndc.nodes.Update(nodeCopy)
+			if errUpdate != nil {
+				return node, fmt.Errorf("failed to update node to clean up annotations: %w", errUpdate)
```

Comment:
```suggestion
				return node, errors.Join(fmt.Errorf("failed to update node to clean up annotations: %w", errUpdate), err)
```
I think we should include the `err` in this returned error so that the original reason for the drain failures is retained.

Reply:
Right, but i suggest to use a different order of the errros: `return node, errors.Join(err, fmt.Errorf("failed to update node to clean up annotations: %w", errUpdate))`.
`err` is the origin error that triggered the whole thing, so it has to be the first mentioned in the concatenated error.